src/tests/Microsoft.PowerFx.Core.Tests/ExpressionTestCases/Patch_PatchRecord.txt

```diff
@@ -0,0 +1,27 @@
+﻿#SETUP: SupportColumnNamesAsIdentifiers
+
+// **********************************************************
+// Patch(Record, Updates1, Updates2,…)
+// **********************************************************
+
+>> Patch({Field1:1,Field2:"earth",Field3:DateTime(2022,1,1,0,0,0,0),Field4:true}, {Field1:2,Field2:"moon",Field3:DateTime(2022,2,1,0,0,0,0),Field4:false})
+{Field1:2,Field2:"moon",Field3:DateTime(2022,2,1,0,0,0,0),Field4:false}
+
+>> Patch({Field1:1,Field2:"earth",Field3:DateTime(2022,1,1,0,0,0,0),Field4:true}, {}, {Field1:2,Field2:"moon",Field3:DateTime(2022,2,1,0,0,0,0),Field4:false})
+{Field1:2,Field2:"moon",Field3:DateTime(2022,2,1,0,0,0,0),Field4:false}
+
+>> Patch(Blank(), {Field1:1,Field2:"earth",Field3:DateTime(2022,1,1,0,0,0,0),Field4:true}, {Field1:2,Field2:"moon",Field3:DateTime(2022,2,1,0,0,0,0),Field4:false})
+{Field1:2,Field2:"moon",Field3:DateTime(2022,2,1,0,0,0,0),Field4:false}
+
+>> Patch({Field1:1,Field2:"earth",Field3:DateTime(2022,1,1,0,0,0,0),Field4:true}, {Field1:2,Field2:"moon",Field3:DateTime(2022,2,1,0,0,0,0),Field4:false}, {ErrorField:1/0})
+{ErrorField:Error({Kind:ErrorKind.Div0}),Field1:2,Field2:"moon",Field3:DateTime(2022,2,1,0,0,0,0),Field4:false}
+
+>> Patch({Field1:1,Field2:"earth",Field3:DateTime(2022,1,1,0,0,0,0),Field4:true}, {Field1:2,Field2:"moon",Field3:DateTime(2022,2,1,0,0,0,0),Field4:false}, 1/0)
+Errors: Error 153-154: Cannot use a non-record value in this context: '1 / 0'.|Error 0-156: The function 'Patch' has some invalid arguments.
+
+// Flattening record fields
```

Comment:
More interesting tests:

- Patching error records (will return error): Patch({a:1},If(1/0<2,{b:3}))
- Patch with incompatible types (it returns a compile-time error): Patch({a:1},{a:[2]})
- Patch with compatible types (I don't know if it works, but it will be good to have the tests to spec what it does):Patch({a:1},{a:"2"}), Patch({a:"1"},{a:2})
- Patch with partial updates: Patch({a:1,b:2},{b:3})
- Patch with multiple partial updates: Patch({a:1,b:"hello",c:false}, {b:"hello 2"},{c:true},{d:[1,2,3]})
- Patch with record columns: Patch({name:"John Doe",address:{street:"One Main St",city:"Springfield",country:"USA"}},{address:{city:"Smallville"}}) --> does it replace the entire 'address' record, or only the nested 'city'? I think the former, but we should specify it as well
- Same with tables instead of records
- Records with blank properties: Patch({a:1,b:"hello"},{b:Blank()})